src/cv/apply_detections.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import argparse
+import imutils
+import json
+import pickle
+import cv2
+import os
+import settings
+import time
+import yolo_video_detect as obj_detector
+import recognize_faces as face_recognize
+#import cv_producer
+#import insert_frame
+
+def face_tracker_helper(face_tracker, new_detections):
+    """ Returns True if there is at least 1 new detection """
+    ret_val = False
+    for i in range(len(new_detections)):
+        if new_detections[i] == "unknown":
+            continue
+        elif not new_detections[i] in face_tracker:
+            face_tracker.append(new_detections[i])
+            ret_val = True
+    return ret_val
+
+def init_video_stream(args):
+    """ Obtains the video stream """
+    stream = cv2.VideoCapture(args["input"])
+    #stream = cv2.VideoCapture(0)
+    try:
+        if imutils.is_cv2():
+            prop = cv2.cv.CV_CAP_PROP_FRAME_COUNT
+        else:
+            prop = cv2.CAP_PROP_FRAME_COUNT
+        total = int(stream.get(prop))
+        print("{} total frames in video".format(total))
+
+    except:
+        raise Exception("Could not determine # of frames in video")
+        total = -1
+
+    return stream
+
+def iterate_frames(args, detector, embedder, recognizer, le, net, ln, colors, labels, total):
+    """ Reading in each frame and have it processed """
+    stream = init_video_stream(args)
```

Comment:
Just like before, what happens if the stream doesn't exist?

Reply:
won't read any frames when grab is called and will exit